src/mito2/src/sst/parquet/reader.rs

```diff
@@ -515,6 +552,100 @@ impl ParquetReader {
         Ok(None)
     }
 
+    /// Prunes batches by the pushed down predicate.
+    fn prune_batches(&mut self) -> Result<()> {
+        // fast path
+        if self.predicate.is_empty() {
+            return Ok(());
+        }
+
+        let mut new_batches = VecDeque::new();
+        let mut batches = std::mem::take(&mut self.batches);
+        for batch in batches.drain(..) {
```

Comment:
The following two patterns should work.

```suggestion
        for batch in self.batches.drain(..) {
```

or
```suggestion
        let mut batches = std::mem::take(&mut self.batches);
        for batch in self.batches {
```